GraphicEditor/View/Building1FirstFloorPlan.xaml.cs

```diff
@@ -0,0 +1,219 @@
+                    rectangle.MouseEnter += PatientRoom_MouseEnter;
+                    rectangle.MouseLeave += PatientRoom_MouseLeave;
+                }
+                else if(rectangle.Name.Equals("ExaminationRoom"))
+                {
+                    rectangle.MouseDown += AdditionalInformationExaminationRoom;
+                    rectangle.MouseEnter += ExaminationRoom_MouseEnter;
+                    rectangle.MouseLeave += ExaminationRoom_MouseLeave;
+                }
+                else if (rectangle.Name.Equals("OperatingRoom"))
+                {
+                    rectangle.MouseDown += AdditionalInformationOperatingRoom;
+                    rectangle.MouseEnter += OperatingRoom_MouseEnter;
+                    rectangle.MouseLeave += OperatingRoom_MouseLeave;
+                }
+                else if (rectangle.Name.Equals("AuxiliaryRoom"))
+                {
+                    rectangle.MouseDown += AdditionalInformationAuxiliaryRoom;
+                    rectangle.MouseEnter += AuxiliaryRoom_MouseEnter;
+                    rectangle.MouseLeave += AuxiliaryRoom_MouseLeave;
+                }
+                else if (rectangle.Name.Equals("StorageRoom"))
+                {
+                    rectangle.MouseDown += AdditionalInformationAuxiliaryRoom;
+                    rectangle.MouseEnter += AuxiliaryRoom_MouseEnter;
+                    rectangle.MouseLeave += AuxiliaryRoom_MouseLeave;
+                }
+            }
+
+        }
+        private void mouseClickArrowUp(object sender, MouseButtonEventArgs e)
+        {
+            page.MainFrame.Content = new Building1SecondFloorPlan();
+            if (page.tabControl.SelectedIndex == 0)
+            {
+                page.comboBoxH1.SelectedIndex = 2;
+            }
+            else
+            {
+                page.comboBoxHospital1.SelectedIndex = 2;
+            }
+        }
+        private void mouseClickArrowDown(object sender, MouseButtonEventArgs e)
+        {
+            page.MainFrame.Content = new Building1FloorPlan();
+            if (page.tabControl.SelectedIndex == 0)
+            {
+                page.comboBoxH1.SelectedIndex = 0;
+            }
+            else
+            {
+                page.comboBoxHospital1.SelectedIndex = 0;
+            }
+        }
+
+        private void PatientRoom_MouseEnter(object sender, MouseEventArgs e)
+        {
+            if (ChooseUser.Restriction == 0)
+            {
+                PopupPatientRoom.Placement = PlacementMode.MousePoint;
+                PopupPatientRoom.IsOpen = true;
+            }
+            else
+            {
+                PopupInfoForPatient.Placement = PlacementMode.MousePoint;
+                PopupInfoForPatient.IsOpen = true;
+            }
+        }
+
+        private void PatientRoom_MouseLeave(object sender, MouseEventArgs e)
+        {
+            PopupPatientRoom.Visibility = Visibility.Collapsed;
```

Comment:
Try to avoid to multiply similar lines,you can use multiple assign statements
https://www.codegrepper.com/code-examples/csharp/assign+multiple+variables+in+one+line+c%23

Reply:
agree